### ISSUE TITLE
Retain text ranges for configuration values read from PEP 723 metadata

### ISSUE THREAD

**Issue by MichaReiser** [astral-sh/ty#4180]
The PEP 723 configuration parser currently doesn't retain the values' source ranges. This prevents diagnostics, like our python version hint, from pointing to the precise source.

E.g. this diagnostic should contain a code frame showing where the python version is configured, but it does not.

```rust
#[test]
fn python_version_diagnostics_identify_script_metadata() -> anyhow::Result<()> {
    let case = CliTest::with_file(
        "script.py",
        r#"
        # /// script
        # requires-python = ">=3.12"
        # ///

        PythonFinalizationError
        "#,
    )?;

    assert_cmd_snapshot!(case.command(), @"
    success: false
    exit_code: 1
    ----- stdout -----
    error[unresolved-reference]: Name `PythonFinalizationError` used when not defined
     --> script.py:6:1
      |
    6 | PythonFinalizationError
      | ^^^^^^^^^^^^^^^^^^^^^^^
    info: `PythonFinalizationError` was added as a builtin in Python 3.13
    info: Python 3.12 was assumed when resolving types because it was specified in script metadata

    Found 1 diagnostic

    ----- stderr -----
    ");
```

This requires some form of source map that maps positions from the extracted PEP 723 metadata header (which is what we pass to the TOML parser) to absolute offsets into the original script.
